app/services/articles/suggest_stickies.rb

```diff
@@ -23,31 +32,37 @@ def call
     def tag_articles
       article_tags = article.cached_tag_list_array - ["discuss"]
 
-      Article
-        .published
-        .tagged_with(article_tags, any: true).unscope(:select)
-        .limited_column_select
+      scope = Article
         .where("public_reactions_count > ? OR comments_count > ?", reaction_count_num, comment_count_num)
-        .where.not(id: article.id)
-        .where.not(user_id: article.user_id)
-        .where("featured_number > ?", 5.days.ago.to_i)
-        .order(Arel.sql("RANDOM()"))
         .limit(3)
+
+      apply_common_scope(scope: scope, tags: article_tags)
     end
 
+    # @note This previously used a `.limit(10 - tag_articles.count)`,
+    #       which meant we were running an unnecessary count query.
+    #       We also had a guard clause that said if we have more than
+    #       6 tag_articles, we should return an empty array.  However,
+    #       with the initial limit of 3 for tag articles, that guard
+    #       was unnecessary.
     def more_articles
-      return [] if tag_articles.size > 6
+      scope = Article
+        .where("comments_count > ?", comment_count_num)
+        .limit(7)
```

Comment:
is this limit  7 = 10 - "initial limit of 3 for tag articles", or something else?

Reply:
It was 10 - initial limit of 3 (or less).  But also we then randomize later.  Almost went with a constant on this.

Reply:
I think I'd actually prefer a constant.

Reply:
Named constant sounds good to me.